pennylane/measurements/shots.py

```diff
@@ -12,14 +12,27 @@
 # limitations under the License.
 """This module contains the Shots class to hold shot-related information."""
 # pylint:disable=inconsistent-return-statements
-from collections import namedtuple
+from typing import NamedTuple
 from typing import Sequence, Tuple
 
-ShotCopies = namedtuple("ShotCopies", ["shots", "copies"])
-"""A namedtuple that represents a shot quantity being repeated some number of times.
 
-For example, ``ShotCopies(shots=10, copies=2)`` indicates two executions with 10 shots each for 20 shots total.
-"""
+class ShotCopies(NamedTuple):
+    """A namedtuple that represents a shot quantity being repeated some number of times.
+    For example, ``ShotCopies(shots=10, copies=2)`` indicates two executions with 10 shots each for 20 shots total.
+    """
+
+    shots: int
+    copies: int
+
+    def __str__(self):
+        """The string representation of the class"""
+        if self.copies > 1:
+            return f"({self.shots} shots x {self.copies})"
+        return f"{self.shots}"
```

Comment:
I think I would prefer to always return the `n shots x copies` format, even if self.copies is 1. Otherwise, you get things like `shot_vector=[1, (10 shots x 2)]` which looks... ragged? I think that's the word I want here. Maybe something slightly different from that, but preferably something consistent.

Reply:
With or without the parenthesis?

Reply:
In truth I'm not super picky here, so I'll just list a bunch and you choose what pleases you. This should remain an internal detail anyway, so I'm not super worried about repr. working with `[(1,1), (10, 5), (100, 1)]`, we could do any of these:
1. `[1, (10 shots x 5), 100]`
2. `[(1 shots x 1), (10 shots x 5), (100 shots x 1)]`
3. `[1 shots x 1, 10 shots x 5, 100 shots x 1]`
4. `[1 shots, 10 shots x 5, 100 shots]`
5. `[1 x 1, 10 x 5, 100 x 1]`
6. `[1x1, 10x5, 100x1]`

You can drop the plural "s" for "1 shot" or not, I don't mind. My personal pick from above is option 4.
`return f"{x.shots} shots{' x '+str(x.copies) if x.copies>1 else ''}"`

Reply:
Added option 4 👍🏼